job-scheduler/src/main/java/org/hawkular/metrics/scheduler/impl/SchedulerImpl.java

```diff
@@ -263,10 +259,10 @@ public void start() {
                             return Observable.just(time);
                         });
                     })
-//                    .doOnNext(activeTimeSlices::remove)
                     .subscribe(
                             d -> {
                                 logger.debug("Finished post job execution clean up for [" + d + "]");
+                                // TODO should this be in a synchronized block?
```

Comment:
I think multiple thread execute the remove concurrently which is why I do't have it in a synchronized block. I wanted you to look at it as well though.